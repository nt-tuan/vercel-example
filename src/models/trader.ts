import { KYCStatus } from "./kyc";

export interface Trader {
  marketplace: string;
  username: string;
  email: string;
  dateJoined: string;
  status: KYCStatus;
}

export interface TraderContact {
  companyName?: string;
  phoneNumber?: string;
  email?: string;
  country?: string;
  address?: string;
  emailVerified: boolean;
}
