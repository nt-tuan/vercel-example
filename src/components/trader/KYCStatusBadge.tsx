import { Chip } from "@mui/material";
import { KYCStatus, KYCStatusLabel } from "../../models/kyc";

const statusColor: {
  [key in KYCStatus]: "error" | "warning" | "info" | "success" | "default";
} = {
  [KYCStatus.FAILED]: "error",
  [KYCStatus.IN_PROGRESS]: "warning",
  [KYCStatus.PENDING_EMAIL]: "info",
  [KYCStatus.PASSED]: "success",
  [KYCStatus.EMAIL_VERIFIED]: "success",
  [KYCStatus.INACTIVE]: "default",
};
interface Props {
  status?: KYCStatus;
  size?: "small" | "medium";
}
export const KYCStatusBadge = ({ status, size }: Props) => {
  if (!status) return <></>;
  return (
    <Chip
      color={statusColor[status]}
      label={KYCStatusLabel[status] ?? "default"}
      size={size}
    />
  );
};
