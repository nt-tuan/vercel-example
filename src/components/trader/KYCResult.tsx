import React from "react";
import { KYCResult as Model, KYCStatus } from "models/kyc";
import { DataGrid, GridColDef, GridValueGetterParams } from "@mui/x-data-grid";
import { IconButton } from "@material-ui/core";
import VisibilityIcon from "@mui/icons-material/Visibility";
import { KYCStatusBadge } from "./KYCStatusBadge";
import { Dialog, DialogContent, DialogTitle } from "@mui/material";
interface Props {
  data?: Model[];
}

export const KYCResult = ({ data }: Props) => {
  const [message, setMessage] = React.useState<string>();
  const handleRowClick = () => {};
  const viewMessage = (message?: string) => {
    setMessage(message);
  };
  const columns: GridColDef[] = [
    {
      field: "id",
      hide: true,
    },
    {
      field: "date",
      headerName: "Date",
    },
    {
      field: "type",
      headerName: "Type",
    },
    {
      field: "provider",
      headerName: " Provider",
      flex: 1,
    },
    {
      field: "status",
      headerName: "Status",
      width: 130,
      renderCell: (params: GridValueGetterParams) => (
        <KYCStatusBadge status={params.value as KYCStatus} />
      ),
    },
    {
      field: "action",
      headerName: "Action",
      renderCell: (params: GridValueGetterParams) => {
        return (
          <IconButton onClick={() => viewMessage(params.row.message)}>
            <VisibilityIcon />
          </IconButton>
        );
      },
    },
  ];
  return (
    <>
      <DataGrid
        rows={data ?? []}
        columns={columns}
        loading={data == null}
        hideFooter
        disableColumnMenu
        autoHeight
        onRowClick={handleRowClick}
      />
      <Dialog
        onClose={() => setMessage(undefined)}
        open={message != null}
        fullWidth
      >
        <DialogTitle>KYC request and response </DialogTitle>
        <DialogContent>{message}</DialogContent>
      </Dialog>
    </>
  );
};
