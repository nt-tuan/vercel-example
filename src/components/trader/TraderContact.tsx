import {
  Card,
  CardContent,
  List,
  ListItem,
  Skeleton,
  Typography,
} from "@mui/material";
import React from "react";
import { TraderContact as Model } from "models/trader";
import { styled } from "@mui/system";
import { KYCStatusBadge } from "./KYCStatusBadge";
import { KYCStatus } from "models/kyc";

const Flex = styled("div")({
  width: "100%",
  display: "flex",
  justifyContent: "space-between",
});
const FlexItem = styled("div")({
  flex: "1 1 0",
  padding: "16px 0 0 0",
});

const WrappedItem = ({
  label,
  value,
}: {
  label: React.ReactNode;
  value: React.ReactNode;
}) => {
  return (
    <ListItem disablePadding>
      <Flex>
        <FlexItem>{label}</FlexItem>
        <FlexItem>{value}</FlexItem>
      </Flex>
    </ListItem>
  );
};

interface Props {
  contact?: Model;
  status?: KYCStatus;
}

export const TraderContact = ({ contact }: Props) => {
  if (contact == null)
    return <Skeleton variant="rectangular" width={500} height={300} />;
  return (
    <Card style={{ maxWidth: "500px" }}>
      <CardContent>
        <Typography fontWeight="bold">Contact Details</Typography>
        <List>
          <WrappedItem value={contact.companyName ?? ""} label="Company Name" />
          <WrappedItem value={contact.phoneNumber ?? ""} label="Phone Number" />
          <WrappedItem
            value={
              <div>
                <div>{contact.email}</div>
                {contact.emailVerified && (
                  <KYCStatusBadge
                    status={KYCStatus.EMAIL_VERIFIED}
                    size="small"
                  />
                )}
              </div>
            }
            label="Email Address"
          />
          <WrappedItem value={contact.country ?? ""} label="Country" />
          <WrappedItem value={contact.address ?? ""} label="Full Address" />
        </List>
      </CardContent>
    </Card>
  );
};
