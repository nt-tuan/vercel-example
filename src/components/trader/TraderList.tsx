import { Trader, TraderStatus } from "models/trader";
import {
  DataGrid,
  GridColDef,
  GridRowParams,
  GridValueGetterParams,
} from "@mui/x-data-grid";
import { TraderStatusBadge } from "./TraderStatusBadge";
import { useHistory } from "react-router-dom";

const columns: GridColDef[] = [
  {
    field: "marketplace",
    headerName: "Marketplace",
    width: 130,
    sortable: false,
    filterable: false,
    resizable: false,
  },
  {
    field: "username",
    headerName: "Username",
    minWidth: 130,
    flex: 1,
    sortable: false,
    filterable: false,
    resizable: false,
  },
  {
    field: "email",
    headerName: "Email",
    width: 130,
    sortable: false,
    filterable: false,
    resizable: false,
  },
  {
    field: "dateJoined",
    headerName: "Date Joined",
    sortable: false,
    filterable: false,
    resizable: false,
    width: 130,
  },
  {
    field: "status",
    headerName: "Status",
    width: 180,
    sortable: false,
    filterable: false,
    resizable: false,
    renderCell: (params: GridValueGetterParams) =>
      params.value && (
        <TraderStatusBadge
          status={
            TraderStatus[params.value.toString() as keyof typeof TraderStatus]
          }
        />
      ),
  },
];

export const TraderList = ({ traders }: { traders?: Trader[] }) => {
  const history = useHistory();
  const handleRowClick = (params: GridRowParams) => {
    history.push({ pathname: "/trader/" + params.row.id });
  };
  return (
    <DataGrid
      rows={traders ?? []}
      columns={columns}
      loading={traders == null}
      hideFooter
      disableColumnMenu
      autoHeight
      onRowClick={handleRowClick}
    />
  );
};
