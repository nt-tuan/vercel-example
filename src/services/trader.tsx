import { Trader, TraderContact } from "models/trader";
import { HttpAPI } from "./httpAPI";

const traderAPI = new HttpAPI("/core-services/dc-support");
export const getTraders: () => Promise<Trader[]> = async () => {
  return traderAPI.get<Trader[]>("/traders.json");
};

export const getTrader = async (id: string) => {
  return traderAPI.get<TraderContact>("/trader.json");
};
