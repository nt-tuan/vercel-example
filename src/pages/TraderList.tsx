import React from "react";
import { TraderList as TraderTable } from "components/trader/TraderList";
import { getTraders } from "services/trader";
import { Trader } from "models/trader";
import { AdminLayout } from "layouts/AdminLayout";

const TraderList = () => {
  const [traders, setTraders] = React.useState<Trader[]>();
  React.useEffect(() => {
    let subscription = true;
    getTraders().then((resposne) => {
      if (!subscription) return;
      setTraders(resposne);
    });
    return () => {
      subscription = false;
    };
  }, []);
  return (
    <div>
      <AdminLayout.Header>Traders</AdminLayout.Header>
      <AdminLayout.Content>
        <TraderTable traders={traders} />
      </AdminLayout.Content>
    </div>
  );
};

export default TraderList;
