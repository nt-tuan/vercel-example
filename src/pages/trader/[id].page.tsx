import React from "react";
import { TraderContact } from "components/trader/TraderContact";
import { TraderContact as Model } from "models/trader";
import { getTrader } from "services/trader";
import { TraderVerifyingTable } from "components/trader/TraderVerifyingTable";
import { AdminLayout, getLayout } from "layouts/AdminLayout";
import { Box, Skeleton, Typography } from "@mui/material";
import { WrappedLink } from "components/common/WrappedLink/WrappedLink";
import { useRouter } from "next/router";

const Page = () => {
  const [trader, setTrader] = React.useState<Model>();
  const router = useRouter();
  const { id } = router.query;

  React.useEffect(() => {
    if (typeof id !== "string") return;
    getTrader(id).then(setTrader);
  }, [id]);
  return (
    <>
      <AdminLayout.Header>
        {trader?.marketplace ?? <Skeleton variant="text" />} |{" "}
        {trader?.companyName ?? <Skeleton variant="text" />}
      </AdminLayout.Header>
      <AdminLayout.Breadcrumbs aria-label="breadcrumb">
        <WrappedLink
          color="inherit"
          underline="hover"
          href="/"
          fontWeight="bold"
        >
          Trader
        </WrappedLink>
        <Typography fontWeight="light">Trader Details</Typography>
      </AdminLayout.Breadcrumbs>
      <TraderContact contact={trader} />
      <Box sx={{ marginTop: "24px" }}>
        <TraderVerifyingTable data={trader?.verifyingResult} />
      </Box>
    </>
  );
};

Page.getLayout = getLayout;
export default Page;
