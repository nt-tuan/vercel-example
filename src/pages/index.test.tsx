import { act, render } from "@testing-library/react";
import * as traderService from "services/trader";
import { TraderStatus } from "models/trader";
import TraderList from "./index.page";
jest.mock("services/trader");
test("TraderList should render", async () => {
  traderService.getTraders = async () => [
    {
      id: "1",
      marketplace: "Hotel Supply B2B",
      username: "tuan",
      email: "tuan@tuan.tuan  ",
      dateJoined: "2021-10-11",
      status: TraderStatus.PASSED,
    },
  ];
  let containerElement;
  await act(async () => {
    const { container } = render(<TraderList />);
    containerElement = container;
  });
  expect(containerElement).toMatchSnapshot();
});
