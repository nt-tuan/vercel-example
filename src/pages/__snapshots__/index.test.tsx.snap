// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TraderList should render 1`] = `
<div>
  <div>
    <h2
      class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
    >
      Traders
    </h2>
    <div
      class="MuiBox-root css-178yklu"
    >
      <div
        aria-colcount="5"
        aria-multiselectable="false"
        aria-rowcount="1"
        class="MuiDataGrid-root MuiDataGrid-root MuiDataGrid-autoHeight"
        role="grid"
      >
        <div>
          <div />
        </div>
        <div
          class="MuiDataGrid-main"
        >
          <div
            class="MuiDataGrid-columnsContainer"
            style="min-height: 56px; max-height: 56px; line-height: 56px;"
          >
            <div
              aria-rowindex="1"
              class="MuiDataGrid-columnHeaderWrapper scroll"
              role="row"
              style="transform: translate3d(0px, 0px, 0px); min-width: 700px;"
            >
              <div
                aria-colindex="1"
                class="MuiDataGrid-columnHeader"
                data-field="marketplace"
                role="columnheader"
                style="width: 130px; min-width: 130px; max-width: 130px;"
                tabindex="0"
              >
                <div
                  class="MuiDataGrid-columnHeaderDraggableContainer"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainer"
                  >
                    <div
                      aria-label=""
                      class="MuiDataGrid-columnHeaderTitle"
                      data-mui-internal-clone-element="true"
                    >
                      Marketplace
                    </div>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator"
                  style="min-height: 56px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                    data-testid="SeparatorIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                aria-colindex="2"
                class="MuiDataGrid-columnHeader"
                data-field="username"
                role="columnheader"
                style="width: 130px; min-width: 130px; max-width: 130px;"
                tabindex="-1"
              >
                <div
                  class="MuiDataGrid-columnHeaderDraggableContainer"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainer"
                  >
                    <div
                      aria-label=""
                      class="MuiDataGrid-columnHeaderTitle"
                      data-mui-internal-clone-element="true"
                    >
                      Username
                    </div>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator"
                  style="min-height: 56px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                    data-testid="SeparatorIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                aria-colindex="3"
                class="MuiDataGrid-columnHeader"
                data-field="email"
                role="columnheader"
                style="width: 130px; min-width: 130px; max-width: 130px;"
                tabindex="-1"
              >
                <div
                  class="MuiDataGrid-columnHeaderDraggableContainer"
                  draggable="false"
                >
                  <div
                    class="MuiDataGrid-columnHeaderTitleContainer"
                  >
                    <div
                      aria-label=""
                      class="MuiDataGrid-columnHeaderTitle"
                      data-mui-internal-clone-element="true"
                    >
                      Email
                    </div>
                  </div>
                </div>
                <div
                  class="MuiDataGrid-columnSeparator"
                  style="min-height: 56px; opacity: 1;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiDataGrid-iconSeparator css-i4bv87-MuiSvgIcon-root"
                    data-testid="SeparatorIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 19V5h2v14z"
                    />
                  </svg>
                </div>
              </div>
              <div
                class="MuiDataGrid-cell"
                style="min-width: 310px; max-width: 310px; line-height: 55px; min-height: 56px; max-height: 56px;"
              />
            </div>
          </div>
          <div
            style="overflow: visible; width: 0px;"
          >
            <div
              class="MuiDataGrid-windowContainer"
              style="width: 0px; height: 108px;"
            >
              <div
                class="MuiDataGrid-window"
                style="top: 56px; overflow-y: hidden;"
              >
                <div
                  class="MuiDataGrid-dataContainer"
                  style="min-width: 700px; min-height: 52px;"
                >
                  <div
                    class="MuiDataGrid-viewport"
                    style="min-width: 0; max-width: 0; max-height: 52px;"
                  >
                    <div
                      class="MuiDataGrid-renderingZone"
                      style="max-height: 52px; width: 700px; transform: translate3d(0px, 0px, 0px);"
                    >
                      <div
                        aria-rowindex="2"
                        aria-selected="false"
                        class="MuiDataGrid-row"
                        data-id="1"
                        data-rowindex="0"
                        role="row"
                        style="max-height: 52px; min-height: 52px;"
                      >
                        <div
                          aria-colindex="1"
                          class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                          data-colindex="0"
                          data-field="marketplace"
                          role="cell"
                          style="min-width: 130px; max-width: 130px; min-height: 52px; max-height: 52px; line-height: 51px;"
                          tabindex="-1"
                        >
                          Hotel Supply B2B
                        </div>
                        <div
                          aria-colindex="2"
                          class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                          data-colindex="1"
                          data-field="username"
                          role="cell"
                          style="min-width: 130px; max-width: 130px; min-height: 52px; max-height: 52px; line-height: 51px;"
                          tabindex="-1"
                        >
                          tuan
                        </div>
                        <div
                          aria-colindex="3"
                          class="MuiDataGrid-cell MuiDataGrid-cell--textLeft"
                          data-colindex="2"
                          data-field="email"
                          role="cell"
                          style="min-width: 130px; max-width: 130px; min-height: 52px; max-height: 52px; line-height: 51px;"
                          tabindex="-1"
                        >
                          tuan@tuan.tuan  
                        </div>
                        <div
                          class="MuiDataGrid-cell"
                          style="min-width: 310px; max-width: 310px; line-height: 51px; min-height: 52px; max-height: 52px;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="Mui-resizeTriggers"
          >
            <div
              class="expand-trigger"
            >
              <div
                style="width: 1px; height: 1px;"
              />
            </div>
            <div
              class="contract-trigger"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
